extensions/Worker.Extensions.Http.AspNetCore/src/HttpUriProvider.cs

```diff
@@ -7,26 +7,18 @@ namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore
 {
     internal static class HttpUriProvider
     {
-        private static Uri? _httpUri;
+        public static Lazy<Uri> HttpUri = new Lazy<Uri>(() => GetHttpUri());
 
         public static Lazy<int> HttpPort = new Lazy<int>(() => Utilities.GetUnusedTcpPort());
 
+        public static Lazy<string> HttpUriString = new Lazy<string>(() => HttpUri.Value.ToString());
+
         public static Uri GetHttpUri()
```

Comment:
let's simplify this API:
- one public member -- `public static string GetHttpUri()` (since we never use the `Uri` -- only the string)
- one private lazy `HttpUriString` that does all the magic of calling `GetUnusedTcpPort()` and building the uri in its factory
- have `GetHttpUri()` return  `HttpUriString.Value`